sql/src/main/java/io/crate/operation/reference/sys/check/checks/NumberOfPartitionsSysCheck.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.reference.sys.check.checks;
+
+import io.crate.metadata.ReferenceInfos;
+import io.crate.metadata.doc.DocSchemaInfo;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.table.SchemaInfo;
+import io.crate.metadata.table.TableInfo;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+
+@Singleton
+public class NumberOfPartitionsSysCheck extends AbstractSysCheck {
+
+    private static final int ID = 5;
+    private static final String DESCRIPTION = "The number of partitions of one or more partitioned" +
+            " tables is greater than 1000";
+
+    private static final int PARTITIONS_THRESHOLD = 1000;
+    private final ReferenceInfos referenceInfos;
+
+    @Inject
+    public NumberOfPartitionsSysCheck(ReferenceInfos referenceInfos) {
+        super(ID, new BytesRef(DESCRIPTION), Severity.MEDIUM);
+        this.referenceInfos = referenceInfos;
+    }
+
+    @Override
+    public boolean validate() {
+        return validateDocTablesPartitioning(extractDocSchemaInfo());
+    }
+
+    private DocSchemaInfo extractDocSchemaInfo() {
+        DocSchemaInfo docSchemaInfo = null;
+        for (SchemaInfo schemaInfo : referenceInfos) {
```

Comment:
implement a getSchemaInfo(String schemaName) from getSchemaInfo(TableIdent ident)  in ReferenceInfos and use that